chain/network/src/peer_manager/tests/tier2.rs

```diff
@@ -0,0 +1,252 @@
+use crate::broadcast;
+use crate::network_protocol::testonly as data;
+use crate::peer_manager::connection_store::STORED_CONNECTIONS_MIN_DURATION;
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::peer_manager::peer_manager_actor::{
+    POLL_CONNECTION_STORE_INTERVAL, RECONNECT_ATTEMPT_INTERVAL, UPDATE_CONNECTION_STORE_INTERVAL,
+};
+use crate::peer_manager::testonly::start as start_pm;
+use crate::peer_manager::testonly::ActorHandler;
+use crate::peer_manager::testonly::Event;
+use crate::tcp;
+use crate::testonly::make_rng;
+use crate::time;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::network::PeerId;
+use near_store::db::TestDB;
+use std::sync::Arc;
+
+async fn check_recent_outbound_connections(pm: &ActorHandler, mut wanted: Vec<PeerId>) {
+    let mut got: Vec<PeerId> = pm
+        .with_state(move |s| async move {
+            s.connection_store
+                .get_recent_outbound_connections()
+                .iter()
+                .map(|c| c.peer_info.id.clone())
+                .collect()
+        })
+        .await;
+
+    wanted.sort();
+    got.sort();
+
+    assert!(got == wanted, "Expected {:?} but got {:?}", wanted, got);
+}
+
+async fn wait_for_connection_closed(events: &mut broadcast::Receiver<Event>) {
+    events
+        .recv_until(|ev| match ev {
+            Event::PeerManager(PME::ConnectionClosed(_)) => Some(()),
+            _ => None,
+        })
+        .await
+}
+
+#[tokio::test]
+async fn test_recent_outbound_connection() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let pm0 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+    let pm1 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+    let pm2 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+
+    let id0 = pm0.cfg.node_id();
+    let id1 = pm1.cfg.node_id();
+
+    tracing::info!(target:"test", "connect pm0 to pm1");
+    pm0.connect_to(&pm1.peer_info(), tcp::Tier::T2).await;
+    pm2.connect_to(&pm0.peer_info(), tcp::Tier::T2).await;
+    clock.advance(STORED_CONNECTIONS_MIN_DURATION);
+
+    tracing::info!(target:"test", "check that pm0 stores the outbound connection to pm1");
+    clock.advance(UPDATE_CONNECTION_STORE_INTERVAL);
+    check_recent_outbound_connections(&pm0, vec![id1.clone()]).await;
```

Comment:
are you sure that you don't have to await some event here? There is no guarantee that the ConnectionStore will be updated yet, right?